### ISSUE TITLE
Mail from must equal authorized user

### ISSUE THREAD

**Issue by lionelmeng** [matomo-org/matomo#12426]
Version:Piwik 3.2.1
When i send a mail report, it occurs the error: Mail from must equal authorized user

Is there something wrong in my mail configuration?

**Comment by Findus23** [comment]
Hi,
I haven't heard of this error message before (and hoped that we wouldn't have such indescriptive error messages).
But it seems like this is a message from your SMTP server, can you check if you used the correct email in the `From` field?

**Comment by nyanim** [comment]
Hello @Findus23 , 

I encountered the same issue and my analysis are as follows:

I captured packets with WireShark when sending an email report. 
![2019-02-26 6 59 39](https://user-images.githubusercontent.com/13297785/53410058-ac325680-39fd-11e9-8f4c-d6d9b9cc9a48.png)
As we can see from the image, matomo uses the domain that matomo instance is running on with the "noreply@" prefix as the "MAIL FROM" field in smtp. 
The SMTP server returned a 5xx error, indicates the mail from address does not match the authenticated user.

I tried to reproduce the error manually with telnet, and got the same error when providing a wrong MAIL FROM address.
![2019-02-26 7 03 25](https://user-images.githubusercontent.com/13297785/53410081-bf452680-39fd-11e9-9372-574164700167.png)

I have fixed this issue and submited PR #14137
